### PR TITLE
[UI]: Refactor UI package components to typescript

### PR DESCRIPTION
**Pull request by cmdcolin**
## Description

Depends on https://github.com/opentargets/ot-ui-apps/pull/102

## Type of change

Dependencies/typescripting, split out from https://github.com/opentargets/ot-ui-apps/pull/101

## How Has This Been Tested?

Manually opening app and clicking around

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,8 +23,10 @@
     "format": "prettier --write \"**/*.{ts,tsx,md}\""
   },
   "devDependencies": {
+    "@types/file-saver": "^2.0.5",
```
Comment (cmdcolin) [inline package.json:26]: this branch is 'rebased' onto #102 hence these 'appconfig' changes. same with the genetics pr

**Comment by cmdcolin** [comment]
merged origin/main :)

note: some of the jsx->tsx changes are not well tracked by git, and appear to be a full "file removed"+"file added". this can be improved using a customized git diff, e.g. run this command from this branch

```
git diff origin/main -M30% -w
```

example output for footer.tsx


```
 
diff --git a/packages/ui/src/Footer.jsx b/packages/ui/src/Footer.tsx
similarity index 46%
rename from packages/ui/src/Footer.jsx
rename to packages/ui/src/Footer.tsx
index 97b8797..a596a54 100644
--- a/packages/ui/src/Footer.jsx
+++ b/packages/ui/src/Footer.tsx
@@ -1,33 +1,36 @@
-import React, { Fragment } from "react";
-import { Grid, makeStyles, Typography } from "@material-ui/core";
+import React from "react";
+import { Grid, Typography } from "@material-ui/core";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { withStyles } from "@material-ui/core/styles";
+import { makeStyles } from "@material-ui/core/styles";
 
 import { Link } from "./Link";
 import { EmailLink } from "./EmailLink";
 
 import PrivateWrapper from "../../../apps/platform/src/components/PrivateWrapper";
 
-const styles = (theme) => ({
+const useStyles = makeStyles((theme) => ({
   footer: {
+    // @ts-ignore
     backgroundColor: theme.palette.footer,
     color: "#fff",
     margin: 0,
     width: "100%",
     padding: 24,
   },
-});
+}));
 
-const linkStyles = () => ({
+const useLinkStyles = makeStyles(() => ({
   iconClass: {
     marginRight: "10px",
   },
   linkContainer: {
     marginBottom: "8px",
   },
-});
+}));
 
-let FooterLink = ({ label, url, classes, icon }) => (
+let FooterLink = ({ label, url, icon }) => {
+  const classes = useLinkStyles();
+  return (
     <Grid item xs={12} className={classes.linkContainer}>
       <Typography color="inherit">
         {url.startsWith("mailto") ? (
@@ -47,8 +50,7 @@ let FooterLink = ({ label, url, classes, icon }) => (
       </Typography>
     </Grid>
   );
-
-FooterLink = withStyles(linkStyles)(FooterLink);
+};
 
 const FooterSectionHeading = ({ children }) => (
   <Grid item xs={12}>
@@ -58,7 +60,7 @@ const FooterSectionHeading = ({ children }) => (
   </Grid>
 );
 
-const socialLinkStyle = () => ({
+const useSocialLinkStyle = makeStyles(() => ({
   iconsContainer: {
     maxWidth: "235px",
   },
@@ -66,10 +68,12 @@ const socialLinkStyle = () => ({
     fontSize: "30px",
     color: "white",
   },
-});
+}));
 
-let FooterSocial = ({ social, classes }) => (
-  <Fragment>
+let FooterSocial = ({ social }) => {
+  const classes = useSocialLinkStyle();
+  return (
+    <>
       <FooterSectionHeading>Follow us</FooterSectionHeading>
       <Grid
         className={classes.iconsContainer}
@@ -84,10 +88,9 @@ let FooterSocial = ({ social, classes }) => (
           </Grid>
         ))}
       </Grid>
-  </Fragment>
+    </>
   );
-
-FooterSocial = withStyles(socialLinkStyle)(FooterSocial);
+};
 
 const useSectionStyles = makeStyles({
   section: {
@@ -95,7 +98,20 @@ const useSectionStyles = makeStyles({
   },
 });
 
-const FooterSection = ({ heading, links, social }) => {
+const FooterSection = ({
+  heading,
+  links,
+  social,
+}: {
+  heading: React.ReactNode;
+  links: {
+    showOnlyPartner: boolean;
+    label: string;
+    icon: unknown;
+    url: string;
+  }[];
+  social?: unknown;
+}) => {
   const classes = useSectionStyles();
   return (
     <Grid
@@ -142,7 +158,9 @@ const FooterSection = ({ heading, links, social }) => {
   );
 };
 
-const Footer = ({ classes, externalLinks }) => (
+const Footer = ({ externalLinks }) => {
+  const classes = useStyles();
+  return (
     <Grid
       className={classes.footer}
       container
@@ -164,5 +182,6 @@ const Footer = ({ classes, externalLinks }) => (
       </Grid>
     </Grid>
   );
+};
 
-export default withStyles(styles)(Footer);
+export default Footer;

```

**Comment by cmdcolin** [comment]
any update on this? i see you opened https://github.com/opentargets/ot-ui-apps/pull/108  but that it looks somewhat orthogonal to this

if you don't actually want community contributions please let me know, but you did put out a 'call for community contributions' and so i thought it would be fun to participate

**Comment by cmdcolin** [comment]
thanks for the reply, and apologies for the snark! definitely take your time with it, and if it helps probably single files can be cherry-picked from the branches for smaller units of change too

**Review by carcruz — APPROVED** [review]
Good to go!

**Pull request merged**